change_wheel_version.py

```diff
@@ -101,26 +125,33 @@ def change_wheel_version(
                 dest_dir / new_slug / f"{old_slug}.data", dest_dir / new_slug / f"{new_slug}.data"
             )
 
-        metadata = dest_dir / new_slug / f"{new_slug}.dist-info" / "METADATA"
+        metadata_path = dest_dir / new_slug / f"{new_slug}.dist-info" / "METADATA"
+        wheel_path = dest_dir / new_slug / f"{new_slug}.dist-info" / "WHEEL"
 
         # This is actually a non-conformant email policy as per
         # https://packaging.python.org/en/latest/specifications/core-metadata/
         # However, it works around this bug in setuptools in cases where the version is really long
         # https://github.com/pypa/setuptools/issues/3808
         max_line_length = 200
         policy = email.policy.compat32.clone(max_line_length=200)
+        parser = email.parser.BytesParser(policy=policy)
         version_str = str(new_version)
         if len(version_str) >= max_line_length:
             raise ValueError(f"Version {version_str} is too long")
 
-        with open(metadata, "rb") as f:
-            parser = email.parser.BytesParser(policy=policy)
+        with open(metadata_path, "rb") as f:
             msg = parser.parse(f)
-
         msg.replace_header("Version", version_str)
-        with open(metadata, "wb") as f:
+        with open(metadata_path, "wb") as f:
             f.write(msg.as_bytes())
 
+        if platform_tag:
+            new_tag = change_platform_tag(wheel_path, platform_tag, parser)
```

Comment:
```suggestion
            # We don't need to sort, because we assume len(parse_tag(platform_tag)) == 1
            new_tag = change_platform_tag(wheel_path, platform_tag, parser)
```